unison-cli/src/Unison/Auth/Tokens.hs

```diff
@@ -54,7 +52,7 @@ performTokenRefresh :: MonadIO m => URI -> Tokens -> m (Either CredentialFailure
 performTokenRefresh discoveryURI (Tokens {refreshToken = currentRefreshToken}) = runExceptT $
   case currentRefreshToken of
     Nothing ->
-      throwError $ (RefreshFailure . Text.pack $ "Unable to refresh authentication, please run " <> patternName IP.authLogin <> " and try again.")
+      throwError $ (RefreshFailure . Text.pack $ "Unable to refresh authentication, please run auth.login and try again.")
```

Comment:
Why this change?

Reply:
Circular imports :'(

Reply:
Ah, I see. Yeah Tokens shouldn't really know about InputPatterns, unless we'd want some application reader that knows. But how does input patterns rely back on the auth client?

Reply:
InputPatterns contain argument types which contain completions which now depend on the authed-client so they can get completions from share.